src/QueryReflection/LazyQueryReflector.php

```diff
@@ -43,4 +43,8 @@ public function getResultType(string $queryString, int $fetchType): ?Type
 
         return $this->reflector->getResultType($queryString, $fetchType);
     }
+
+    public function setupDbaApi(?DbaApi $dbaApi): void
+    {
+    }
```

Comment:
store the `dbaApi` in a local property and pass it via `setupDbaApi` after reflector creation.

might be worth it to extract the reflector creation into a new private method

Reply:
> store the dbaApi in a local property and pass it via setupDbaApi after reflector creation.

done

> might be worth it to extract the reflector creation into a new private method

wasnt exactly sure what do you mean

Reply:
`$this->reflector = ($this->reflectorFactory)();` can be found 2 times in this class.

I think we should extract a method

```
private function createReflector():QueryReflector {
    if (null === $this->reflector) {
            $this->reflector = ($this->reflectorFactory)();
            $this->reflector->setupDbaApi($this->dbaApi);
    }
    return $this->reflector;
}
```

and use it from this 2 call sites

Reply:
Sorry, I overlooked the 2nd usage :( Hopefully done

Reply:
no worries. thx for working on it.